File: clients/typescript/examples/collections-find-and-rerank-include-vectorize.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

(async function () {
  // Find documents
  const cursor = collection.findAndRerank(
    {},
    {
      sort: { $hybrid: "A tree in the woods" },
      projection: { is_checked_out: 1, title: 1 },
    },
  );

  // Iterate over the found documents
  for await (const result of cursor) {
    // Documents will only have the requested fields
    // (plus '_id' by default projection)
    console.log(result.document);
  }
})();

File: clients/python/examples/collections-find-and-rerank-include-vectorize.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Find documents
cursor = collection.find_and_rerank(
    sort={"$hybrid": "A tree in the woods"},
    projection={"is_checked_out": True, "title": True},
)

# Iterate over the found documents
for result in cursor:
    # Documents will only have the requested fields
    # (plus '_id' by default projection)
    print(result.document)
